From 26b8224b8bd46a3a56d424ae8b9716c495e597d4 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Wed, 26 Aug 2015 13:13:33 +0000
Subject: [PATCH] Follow-up to BCEL-209 Drop the field num_bootstrap_arguments
 it is not necessary, and was not kept in sync if the arguments were updated
 Oops - incomplete fix, now working

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/bcel/trunk@1697925 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/bcel6/classfile/BootstrapMethod.java     | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/main/java/org/apache/commons/bcel6/classfile/BootstrapMethod.java b/src/main/java/org/apache/commons/bcel6/classfile/BootstrapMethod.java
index 0b031f32a2..2f89639d94 100644
--- a/src/main/java/org/apache/commons/bcel6/classfile/BootstrapMethod.java
+++ b/src/main/java/org/apache/commons/bcel6/classfile/BootstrapMethod.java
@@ -46,7 +46,7 @@ public class BootstrapMethod implements Cloneable {
      * Initialize from another object.
      */
     public BootstrapMethod(BootstrapMethod c) {
-        this(c.getBootstrapMethodRef(), c.getNumBootstrapArguments(), c.getBootstrapArguments());
+        this(c.getBootstrapMethodRef(), c.getBootstrapArguments());
     }
 
     /**
@@ -56,22 +56,24 @@ public BootstrapMethod(BootstrapMethod c) {
      * @throws IOException
      */
     BootstrapMethod(DataInput input) throws IOException {
-        this(input.readUnsignedShort(), input.readUnsignedShort(), (int[]) null);
+        this(input.readUnsignedShort(), input.readUnsignedShort());
 
         for (int i = 0; i < bootstrap_arguments.length; i++) {
             bootstrap_arguments[i] = input.readUnsignedShort();
         }
     }
 
+    // helper method
+    private BootstrapMethod(int bootstrap_method_ref, int num_bootstrap_arguments) {
+        this(bootstrap_method_ref, new int[num_bootstrap_arguments]);
+    }
 
     /**
      * @param bootstrap_method_ref int index into constant_pool of CONSTANT_MethodHandle
-     * @param num_bootstrap_arguments int count of number of boostrap arguments
      * @param bootstrap_arguments int[] indices into constant_pool of CONSTANT_<type>_info
      */
-    public BootstrapMethod(int bootstrap_method_ref, int num_bootstrap_arguments, int[] bootstrap_arguments) {
+    public BootstrapMethod(int bootstrap_method_ref, int[] bootstrap_arguments) {
         this.bootstrap_method_ref = bootstrap_method_ref;
-        this.bootstrap_arguments = new int[num_bootstrap_arguments];
         this.bootstrap_arguments = bootstrap_arguments;
     }
 
